### PR TITLE
Update examples to execute from the root of the repo

### LINKED ISSUE #1671: [DOC]: Update examples to run from project root
### How would you describe the priority of this documentation request

Low (would be nice)

### Please provide a link or source to the relevant docs

examples/README.md

### Describe the problems in the documentation

This would make it easier when performing manual testing of running examples

### (Optional) Propose a correction

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open documentation issues](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Adoc) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description
* Update top-level examples and Triton start-up commands to execute from the root of the repo
* Where possible set default values for cli flags, removing the need to set them for the common use-case
* Where possible remove the need for defining `MORPHEUS_ROOT`
* Ensure C++ Triton pipelines use port 8000 to avoid the warning about the grpc port.
* Optionally cast types in the C++ impl of the Triton stage when `force_convert_inputs=true` and the input and model types didn't match (previously types were always casted)
* Remove `--num_threads=1` restriction and configure logging for the `log_parsing` example
* Remove `--num_threads=8` restriction from `nlp_si_detection` since the pipeline has more than 8 stages.
* Don't invoke the C++ impl of preallocate if the type being requested isn't supported on the C++ side (strings)
* Don't use the C++ impl of the Triton stage if `use_shared_memory` is requested as this isn't supported in C++.
* Add missing `gnn-fraud-classification` stage to CLI alternative for `gnn_fraud_detection_pipeline` example

Closes #1671

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

File: examples/nlp_si_detection/README.md [hunk 1]
```diff
@@ -117,7 +116,7 @@ morpheus --log_level=DEBUG \
    `# 3rd Stage: Preprocessing converts the input data into BERT tokens` \
    preprocess --vocab_hash_file=data/bert-base-uncased-hash.txt --do_lower_case=True --truncation=True \
    `# 4th Stage: Send messages to Triton for inference. Specify the model loaded in Setup` \
-   inf-triton --model_name=sid-minibert-onnx --server_url=localhost:8000 --force_convert_inputs=True \
+   inf-triton --model_name=sid-minibert-onnx --server_url=localhost:8000 \
```
Comment (efajardo-nv) [inline examples/nlp_si_detection/README.md:119]: Getting this error when run command:
```
RuntimeError: No conversion found from mrc::pymrc::PyObjectHolder to std::shared_ptr<morpheus::MultiResponseMessage>
```

File: morpheus/_lib/include/morpheus/objects/dtype.hpp [hunk 2]
```diff
@@ -111,6 +111,10 @@ struct DType
     // From triton
     static DType from_triton(const std::string& type_str);
 
+    // Returns true if the type is fully supported, that is the type can be fully represented as both a triton type and
+    // a cudf type.
+    bool is_fully_supported() const;
+
```
Comment (mdemoret-nv) [inline morpheus/_lib/include/morpheus/objects/dtype.hpp:117]: We should be using the doxygen string format here

File: morpheus/stages/inference/triton_inference_stage.py [hunk 3]
```diff
@@ -756,7 +756,8 @@ def __init__(self,
 
     def supports_cpp_node(self) -> bool:
         # Get the value from the worker class
-        return TritonInferenceWorker.supports_cpp_node()
+        # The C++ impl of this stage does not support the use_shared_memory options
+        return (TritonInferenceWorker.supports_cpp_node() and not self._use_shared_memory)
```
Comment (mdemoret-nv) [inline morpheus/stages/inference/triton_inference_stage.py:760]: Generate a warning message to show that the C++ impl isnt being used. Lets make a bug to fix this in the future.

**Review by mdemoret-nv — APPROVED** [review]
Instead of removing the `--force_convert_inputs` lets pass the variable down and check if its needed. If they dont match, either throw an exception or pass to triton unchanged. Follow the python implementation.

File: examples/nlp_si_detection/README.md (hunk 1 above)
Comment (efajardo-nv) [reply examples/nlp_si_detection/README.md:119]: Works after updating my environment.

**Pull request merged**